sklearn/preprocessing/data.py

```diff
@@ -2108,3 +2109,30 @@ def _transform(self, X):
             for i in range(len(self.transformed_features_)))
         output = np.concatenate([o[..., np.newaxis] for o in outputs], axis=1)
         return output
+
+    def inverse_transform(self, X, y=None):
+        """Invere transform each feature using the lambdas evaluated during fit time
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The transformed data after boxcox transform.
+
+        Returns
+        -------
+        X_inv : array-like, shape (n_samples, n_features)
+            The original data.
+        """
+        
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES, copy=self.copy)
+        if X.shape[1] != self.n_features_:
+            raise ValueError("X has a different shape than during fitting.")
+        X_inv = _transform_selected(X, self._inverse_transform,
+                                    self.transformed_features_, copy=False,
+                                    order=True)
+        return X_inv
+
+    def _inverse_transform(self, X):
+        for i in range(len(self.transformed_features_)):
+            X[:, i] = special.inv_boxcox(X[:, i], self.lambdas_[i])    
```

Comment:
`inv_boxcox` is trivial to implement. Even if this might be a little faster, I don't think that needs to concern us.

You can also do it without iterating over each feature in Python (vectorized across features), something like:
```
Xt = X.copy()
mask = self.lambdas_ != 0
mask_lambdas = self.lambdas_[mask]
Xt_mask = Xt[:, mask]
Xt_mask *= mask_lambdas
np.log1p(Xt_mask, out=Xt_mask)
Xt_mask /= mask_lambdas
np.exp(Xt_mask, out=Xt_mask)
Xt[:, mask] = Xt_mask

mask = self.lambdas_ == 0
Xt[:, mask] = np.exp(Xt[:, mask])
```